### PR TITLE
Add helm charts

### LINKED ISSUE #6: Create helm charts for deploying an instance

### PR DESCRIPTION
**Pull request by anikachurilova**
* initialize helm charts for deploying demo ils instance.
* closes https://github.com/inveniosoftware/demo-invenioils/issues/6

### CODE REVIEW

**Review by ppanero — COMMENTED** [review]
I'm not familiar with the ILS roadmap so I might be saying something completely wrong here... but this seems to be duplicating much of the code present in helm-invenio. Meaning, double maintenance.

Is this meant to be something completely isolated of Invenio helm charts? Would it be worth investigating the [subcharts](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/) to try to use inheritance with the invenio charts?

**Comment by avivace** [comment]
> I'm not familiar with the ILS roadmap so I might be saying something completely wrong here... but this seems to be duplicating much of the code present in helm-invenio. Meaning, double maintenance.
> 
> Is this meant to be something completely isolated of Invenio helm charts? Would it be worth investigating the [subcharts](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/) to try to use inheritance with the invenio charts?

I agree with @ppanero here, we should strive to keep a central repository for them, possibly using subcharts. I briefly checked, but here it looks like some pods/functionality were disabled and custom values applied (e.g. the images), which in theory we should be able to accomplish adding more configuration/guards in the main helm charts

**Comment by ppanero** [comment]
Discussed IRL, service decision. Can go ahead if PR is works.

File: helm-charts/templates/configurations/nginx.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nginx-config
+data:
+  invenio.conf: |
```
Comment (ntarocco) [inline helm-charts/templates/configurations/nginx.yaml]:
```suggestion
  nginx.conf: |
```

File: helm-charts/README-OpenShift.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+## Secret management
+
+Invenio secret key:
+
+```console
+$ SECRET_KEY=$(openssl rand -hex 8)
+$ CSRF_SECRET_SALT=$(openssl rand -hex 8)
+$ SECURITY_LOGIN_SALT=$(openssl rand -hex 8)
+$ oc create secret generic \
+  --from-literal="INVENIO_SECRET_KEY=$SECRET_KEY" \
```
Comment (ntarocco) [inline helm-charts/README-OpenShift.md]: Remove `INVENIO_` from the 3 secrets here:
```suggestion
  --from-literal="SECRET_KEY=$SECRET_KEY" \
```

File: helm-charts/templates/configurations/secrets.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+kind: Secret
+type: Opaque
+metadata:
+  name: invenio-secrets
+  labels:
+    app: invenio-secrets
+  annotations:
+    helm.sh/resource-policy: keep
+data:
+  INVENIO_SECRET_KEY: {{ .Values.invenioils.secret_key | b64enc }}
```
Comment (ntarocco) [inline helm-charts/templates/configurations/secrets.yaml]: 3 times
```suggestion
  SECRET_KEY: {{ .Values.invenioils.secret_key | b64enc }}
```

**Pull request merged**